### PR TITLE
Support symlink layer in image import.

### PR DESCRIPTION
**Pull request by Random-Liu**
Fixes https://github.com/containerd/cri/issues/638.

This PR added support for symlink layers. @AkihiroSuda Does this make sense to you?

Now `ctr cri load` works for `k8s.gcr.io/pause-amd64:3.0`:
```console
$ sudo ctr cri load test/test.tar 
Loaded image: k8s.gcr.io/pause-amd64:3.0
```
/cc @hmtai @qiujian16

Signed-off-by: Lantao Liu <lantaol@google.com>

### CODE REVIEW

File: pkg/containerd/importer/importer.go [hunk 1]
```diff
@@ -60,6 +60,15 @@ func isLayerTar(name string) bool {
 	return slashes == 2 && strings.HasSuffix(name, "/layer.tar")
 }
 
+// followSymlinkLayer returns actual layer name of the symlink layer.
+func followSymlinkLayer(name string) (string, error) {
```
[the remark this answers is not in the document]
Comment (Random-Liu) [reply pkg/containerd/importer/importer.go]: How about this?

[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply pkg/containerd/importer/importer.go]: SGTM but please add a comment about this to the source?

[the remark this answers is not in the document]
Comment (qiujian16) [reply pkg/containerd/importer/importer.go]: We already check with isLayerTar in line 126, maybe remove the check there?

[the remark this answers is not in the document]
Comment (Random-Liu) [reply pkg/containerd/importer/importer.go]: @AkihiroSuda will do.

[the remark this answers is not in the document]
Comment (Random-Liu) [reply pkg/containerd/importer/importer.go]: Done

File: pkg/containerd/importer/importer.go [hunk 2]
```diff
@@ -60,6 +60,22 @@ func isLayerTar(name string) bool {
 	return slashes == 2 && strings.HasSuffix(name, "/layer.tar")
 }
 
+// followSymlinkLayer returns actual layer name of the symlink layer.
+// It returns "deadbeeddeadbeef/layer.tar" if the name is like
```
Comment (mikebrow) [inline pkg/containerd/importer/importer.go]: nit.. the comments sometimes say deadbeeddeadbeef and sometimes say deadbeefdeadbeef replacing these with foo would take less space and hold the same meaning.

0xDEADBEEF holds historical significance as a marker for freed memory in a heap... if you are pointing to deadbeaf it means you have a bad pointer.

File: pkg/containerd/importer/importer.go [hunk 3]
```diff
@@ -60,6 +60,22 @@ func isLayerTar(name string) bool {
...
+// "../deadbeeddeadbeef/layer.tar", and returns error if the name
+// is not in "../xxxx/layer.tar" format.
```
Comment (mikebrow) [inline pkg/containerd/importer/importer.go]: ../foo/layer.tar not xxxx unless you want to use xxxx as the replacement for deadbeeddeadbeef & deadbeefdeadbeef strings...

**Comment by Random-Liu** [comment]
@mikebrow Done.

**Pull request merged**